### DISCUSSION TITLE
Should we use GitHub Release feature?

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#177] [Q&A]
Now that we have the ability to publish AaC to PyPI and we're evaluating a potential VS Code extension for AaC, should we standardize our release process using [GitHub Releases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)?

We currently publish to PyPI using a manual workflow.  This is a good first step and it works.  We could improve this by [triggering the workflow with a GitHub release](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release).  The potential advantage of this approach is it gives us an opportunity to better document our releases for users and ensure we keep our versions synchronized if/when we publish multiple artifacts to multiple repositories.

**Comment by Coffee2Bits** [comment]
I much prefer gating the release/deployment process behind the Github Release trigger than the manual one. I love the idea of providing clearer releases and notes as well. 

That being said, before start producing change logs and versions, we should identify what our version scheme, release cadence, and changelog structure should be.

For the version scheme, how do we want to handle version bumps? Should we assume all pull requests into main are at least a patch version bump unless otherwise specified? We can leverage automated tagging and version bumping github actions to manage most all of this behavior for us.

For the release cadence, I don't have a strong preference. At minimum we should release after every sprint. I'm always happy to release more often, but that could have impacts like making the changelog harder to parse because it's split over more intervals. 

Changelogs are always a pain to prepare for releases, especially if you have several features/changes/pull requests to review; I think we could address this by making sure that when a Pull Request is merged, the commit message contains clear changes/bullets that will be incorporated into the changelog for the release. We can even make the changelog items part of the review by having the author add them to the PR message for reviewers to take a look and critique.

**Reply by cvchaparro** [reply]
I want to second the comment about changelogs. Rather than only having them in the commit messages, I think changelogs can be made far easier to manage/create if we have a clear list of changes made in PR descriptions, as well. If we do this, we have a dedicated place to look for the changes made in a PR and we could even build some tooling around the GitHub API to automatically generate changelogs by pulling the lists of changes from PR descriptions. Just my $0.02.
